packages/flutter/lib/src/material/carousel.dart

```diff
@@ -636,21 +676,40 @@ class _CarouselViewState extends State<CarouselView> {
         _itemExtent = widget.itemExtent == null
             ? null
             : clampDouble(widget.itemExtent!, 0, mainAxisExtent);
-
-        return Scrollable(
-          axisDirection: axisDirection,
-          controller: _controller,
-          physics: physics,
-          viewportBuilder: (BuildContext context, ViewportOffset position) {
-            return Viewport(
-              cacheExtent: 0.0,
-              cacheExtentStyle: CacheExtentStyle.viewport,
-              axisDirection: axisDirection,
-              offset: position,
-              clipBehavior: Clip.antiAlias,
-              slivers: <Widget>[_buildSliverCarousel(theme)],
-            );
+        return NotificationListener<ScrollNotification>(
+          onNotification: (ScrollNotification notification) {
+            if (notification.depth == 0 &&
+                widget.onIndexChanged != null &&
+                notification is ScrollUpdateNotification) {
+              final BuildContext? scrollContext = notification.context;
+              if (scrollContext == null) {
+                return false;
+              }
+              final ScrollPosition position = Scrollable.of(scrollContext).position;
```

Comment:
Why do we need to look up the Scrollable immediately below us? Could the position be accessed using the scroll controller?